Add Textarea styles/examples

KerimRahman opened the pull request:
Part 2 of https://github.com/compassion-global-experience/compassion-design-system/pull/136

KerimRahman committed:
refactor(storybook): update preview background color

KerimRahman committed:
feat(css): add TextField styles/example

KerimRahman committed:
refactor(css): update disabled styles

KerimRahman committed:
refactor(css/form): update error styles

KerimRahman committed:
feat(css): add TextArea styles

kidroca commented:
@KerimRahman 
Could you rebase from `development` to try and fix the merge conflict,
I think it's because I pushed something to the previous PR...

KerimRahman committed:
refactor(css): merge conflicts

# Conflicts:
#	packages/css/src/components/Form/TextField.stories.js
#	packages/css/src/components/Form/form.css

kidroca commented on packages/css/src/components/Form/form.css:
```diff
@@ -13,6 +13,7 @@
 
 .form-input {
   display: block;
+  margin: 4px 0;
```
Do we keep track of these non `var` usages

KerimRahman replied:
Nope, we can create a task once this is merged.

kidroca commented on packages/css/src/components/Form/form.css:
```diff
@@ -28,6 +29,10 @@
   color: var(--form-placeholder-color);
 }
 
+.form-input:focus {
+  border-color: var(--form-focus-border-color);
+}
+
```
For me this is overshadowed by the input outline, which is also bluish

KerimRahman replied:
Should be fine now.

kidroca committed:
refactor(css/TextArea.stories): update symbol counter

kidroca commented on packages/css/src/components/Form/form.css:
```diff
@@ -122,3 +127,20 @@
+}
+
+textarea.form-input {
+  max-width: 100%;
```
What's the effect of `max-width` here? `.form-input` already have `width: 100%` 

Shouldn't we set `min-width` to prevent things like:
<img width="403" alt="Screenshot 2023-01-27 at 13 37 40" src="https://user-images.githubusercontent.com/12156624/215077369-287e3de2-3489-4cba-a4b8-a14a3085c6ad.png">

KerimRahman replied:
The max-width is to prevent dragging outside of the window/parent element. F.x.:
![textarea-max-width-example](https://user-images.githubusercontent.com/4611189/215083090-cafae18c-5591-458a-93cd-03a897637b02.png)

kidroca reviewed (commented):
I've got a few questions

KerimRahman committed:
refactor(css/form): update focus styles

KerimRahman committed:
Merge remote-tracking branch 'origin/form-textarea' into form-textarea

kidroca commented on packages/css/src/components/Form/TextArea.stories.js:
```diff
@@ -0,0 +1,92 @@
+    },
+    state: {
+      control: { type: 'select' },
+      options: ['default', 'disabled', 'active', 'error', 'success'],
```
What does the `active` state do? Is `default` and `active` the same thing?

KerimRahman replied:
Leftover.
